Xamarin.Forms.Core/TouchGestureRecognizer.cs

```diff
@@ -0,0 +1,223 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace Xamarin.Forms
+{
+	public class TouchGestureRecognizer : GestureRecognizer
+	{
+		public static readonly BindableProperty UseVisualStateManagerProperty =
+			BindableProperty.Create(nameof(UseVisualStateManager), typeof(bool), typeof(TouchGestureRecognizer), true);
+
+		public static readonly BindableProperty StateProperty =
+			BindableProperty.Create(nameof(State), typeof(TouchState), typeof(TouchGestureRecognizer), TouchState.Default);
+
+		public static readonly BindableProperty TouchCountProperty =
+			BindableProperty.Create(nameof(TouchCount), typeof(int), typeof(TouchGestureRecognizer), 0);
+
+		readonly Dictionary<int, Touch> _touches = new Dictionary<int, Touch>();
+
+		public TouchState State
+		{
+			get => (TouchState)GetValue(StateProperty);
+			set => SetValue(StateProperty, value);
+		}
+
+		public int TouchCount
+		{
+			get => (int)GetValue(TouchCountProperty);
+			private set => SetValue(TouchCountProperty, value);
+		}
+
+		public List<Touch> Touches => _touches.Values.ToList();
+
+		public bool UseVisualStateManager
+		{
+			get => (bool)GetValue(UseVisualStateManagerProperty);
+			set => SetValue(UseVisualStateManagerProperty, value);
+		}
+
+		public View View { get; private set; }
```

Comment:
I don't think we need to expose the View, as the View will be used as the `sender` for event, or can be configured as CommandParameter for commands

Reply:
done